Issues with jhipster manual upgrade: .yo-rc.json not updated

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

- [ ] Jhipster version is not getting updated in .yo-rc.json file after manual upgrade to master
- [x] Compilation errors after upgrade due to presence of ProfileInfoResource and associated test
- [x] Liquibase checksum issues after upgrade -- Not implemented
- [x] Client side build fails after upgrade due to old files

<!-- Explain the bug, if an error is being thrown a stack trace helps -->

##### **Motivation for or Use Case**
Jhipster version should be updated to ensure correct upgrade path for future upgrades
<!-- Explain why this is a bug for you -->

##### **Reproduce the error**
1. Generate monolith application with Jhipster version 4.14.5.
2. Link "`generator-jhipster`" master codebase to new generated application.
3. Enter `jhipster` command to upgrade to newer version
4. Check jhipster version in `.yo-rc.json`. It still points to `4.14.5`

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Version(s)**
v5.1.0
<!--
Which version of JHipster are you using, is it a regression?
-->

##### **JHipster configuration**

<!--
To provide all information we need, you should run `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
-->
```
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "4.14.5",
    "baseName": "rmono4",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "<secret>",
    "clientFramework": "angularX",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ]
  }
}
```
##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->

##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

- [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

I am also seeing compilation errors after manual upgrade:

> /home/vishal/amono/src/main/java/com/mycompany/myapp/web/rest/ProfileInfoResource.java:[37,62] cannot find symbol
> [ERROR]   symbol:   method getRibbon()
> [ERROR]   location: variable jHipsterProperties of type io.github.jhipster.config.JHipsterProperties
> 

I don't see ribbon property (in `application.yml`) after upgrade, but, java code is still referring to it.

--- a/generators/languages/index.js
+++ b/generators/languages/index.js
@@ -135,6 +135,17 @@ module.exports = class extends BaseGenerator {
         }
     }
 
+    get configuring() {
+        return {
+            saveConfig() {
+                if (this.enableTranslation) {
+                    this.languages = _.union(this.currentLanguages, this.languagesToApply);
+                    this.config.set('languages', this.languages);
+                }
+            }
+        };
+    }
+
     get default() {
         return {
             insight() {
@@ -174,13 +185,6 @@ module.exports = class extends BaseGenerator {
                 }
                 if (configOptions.clientFramework) {
                     this.clientFramework = configOptions.clientFramework;
-                }
-            },
-
-            saveConfig() {
-                if (this.enableTranslation) {
-                    this.languages = _.union(this.currentLanguages, this.languagesToApply);
-                    this.config.set('languages', this.languages);
                 }
             }
         };

